### PR TITLE
bug-1870734: update eliot to python 3.11

### PR DESCRIPTION
**Pull request by relud**

### CODE REVIEW

File: tests/test_cache_manager.py [hunk 1]
```diff
@@ -445,7 +445,7 @@ def test_nested_directories(cm_client, tmpdir):
                                 "num_unhandled_errors": "0",
                                 "event": ["1", "256", "0", "'xul__4byte.symc'"],
                                 "event_flags": ["<flags.CREATE: 256>"],
-                                "flags_list": "'flags.CREATE'",
+                                "flags_list": "'256'",
```
Comment (relud) [inline tests/test_cache_manager.py:448]: enum serialization behavior changed in 3.11 https://docs.python.org/3/whatsnew/3.11.html#enum

File: pyproject.toml [hunk 2]
```diff
@@ -43,6 +43,8 @@ addopts = "-rsxX --showlocals --tb=native -p no:cacheprovider --import-mode=impo
 # Transform all warnings into errors
 filterwarnings = [
     "error",
+    # Falcon currently uses cgi which is going away in python 3.13
+    "ignore:'cgi' is deprecated and slated for removal in Python 3.13:DeprecationWarning",
```
Comment (relud) [inline pyproject.toml:47]: Falcon is planning to fix this in version 4 https://github.com/falconry/falcon/issues/2066, but it supports python 3.11 as of [Falcon 3.1.1](https://github.com/falconry/falcon/releases/tag/3.1.1)

For now python 3.13 isn't coming out for another 11 months, and @willkg is investigating switching to another micro-framework such as flask or fastapi

File: docker/images/fakesentry/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM python:3.10.8-alpine3.16@sha256:d17cfece24cb5d0432b37c138f307a19dd461b88aded1fc6981ef5c997c74de1
+FROM python:3.11.7-alpine3.19@sha256:84271da1cd297b01dd4706e01e7789b08b54a5a512c0e3fcaf11c902640f5ebd
```
Comment (willkg) [inline docker/images/fakesentry/Dockerfile:1]: I keep forgetting to ditch alpine here and everywhere else we use Kent. I updated the Kent suggested Dockerfile to switch to bookworm (Debian 12):

https://github.com/willkg/kent#running-in-a-docker-container

We can keep this and I can toss an item in my todo list to switch kent to use bookworm.

**Review by willkg — APPROVED** [review]
Thank you for doing this!

File: pyproject.toml (hunk 2 above)
Comment (vytas7) [reply pyproject.toml:47]: Hi @relud, yes, we will release either 4.0 or another microrelease to make the framework work on the latest versions of CPython and PyPy.
I understand the wish to consolidate around a smaller number of frameworks, but in the case you reconsider, we are always open to feedback wrt "some pitfalls along the way" :wink:

**Pull request merged**